sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2623,7 +2623,8 @@ def teardown_no_lock(self,
 
                     query_cmd = (
                         f'gcloud compute tpus tpu-vm list --filter='
-                        f'\\(labels.ray-cluster-name={cluster_name}\\) '
+                        f'"(labels.ray-cluster-name={cluster_name} AND '
+                        f'state!=PREEMPTED)" '
```

Comment:
What happens to preempted TPUs? Is it true that they require no further cleanup actions -- i.e., `gcloud compute tpus tpu-vm delete` -- from us (e.g., disks)? Worth a comment!

Reply:
good point I just updated some comments.

According to [reference](https://cloud.google.com/tpu/docs/setup-persistent-disk#set-up-pd-with-tpu), TPU VM's disk is not persistent unless manually specifying it.
So a preempted TPU VM will cost zero and should be safe to ignore for cost perspective. (IIRC, GCP will clean them up after a while)

However, I agree we should try to cleanup the preempted VMs if possible, which requires changing the logic in spot controller to be tearing down VM first and then `sky launch` again.

For now I just wanted to ship to wilson asap. I'll create another PR to fix this.